.github/workflows/build.yml

```
@@ -0,0 +1,129 @@
+name: Build and upload to PyPI
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build_wheels:
+    name: Build ${{ matrix.build }} wheels on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-20.04
+            build: "cp39-*x86_64*"
+          - os: ubuntu-20.04
+            build: "cp39-*i686*"
+          - os: windows-2019
+            build: "cp39-*win_amd64*"
+          - os: windows-2019
+            build: "cp39-*win32*"
+          - os: macos-10.15
+            build: "cp39-*x86_64*"
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # required for versioneer to find tags
+
+      - name: Install build dependencies
+        if: runner.os == 'Windows' || runner.os == 'macOS'
+        run: |
+          pipx install cmake
+          pipx install ninja
+          cmake --version
+          ninja --version
+
+      - name: Build wheels
+        uses: joerick/cibuildwheel@v1.11.0
+        env:
+          CIBW_BUILD: "${{ matrix.build }}"
+          CIBW_ENVIRONMENT_MACOS: "MACOSX_DEPLOYMENT_TARGET=10.10"
+          CIBW_ENVIRONMENT_LINUX: "SKBUILD_CONFIGURE_OPTIONS='-DOPENSSL_ROOT_DIR:PATH=/usr/local/ssl -DCMAKE_JOB_POOL_COMPILE:STRING=compile -DCMAKE_JOB_POOL_LINK:STRING=link -DCMAKE_JOB_POOLS:STRING=compile=2;link=1'"
+          CIBW_MANYLINUX_X86_64_IMAGE: dockcross/manylinux1-x64
+          CIBW_MANYLINUX_I686_IMAGE: dockcross/manylinux1-x86
+          CIBW_BEFORE_BUILD: "pip install -r requirements-repair.txt"
+          CIBW_REPAIR_WHEEL_COMMAND: "python scripts/repair_wheel.py -w {dest_dir} {wheel}"
+          CIBW_TEST_REQUIRES: "-r requirements-test.txt"
+          CIBW_TEST_COMMAND: "pytest {project}/tests"
+
+      - uses: actions/upload-artifact@v2
+        with:
+          path: ./wheelhouse/*.whl
+
+  build_sdist:
+    name: Build source distribution
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0  # required for versioneer to find tags
+
+      - name: Build SDist
+        run: pipx run build --sdist
+
+      - uses: actions/upload-artifact@v2
+        with:
+          path: dist/*.tar.gz
+
+  test_sdist:
+    name: Test SDist with python ${{ matrix.python }}
+    needs: [build_sdist]
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        python: ["2.7", "3.6", "3.10.0-beta.1"]
```

```suggestion
        python: ["2.7", "3.6", "3.10-dev"]
```

Would this be better than updating it every time?

---

I used "3.10" at first which failed and gave me a link "Check here for supported versions". I was hopping to find 3.10-dev or 3.10-nightly or something like that but that was not available.
I just re-read setup-python doc and seems you can specify a range. I'll try that.

---

You can use "3.10-dev" as well, I do that here: https://github.com/pypa/manylinux/blob/16a0a93b90e7d123c38cf1dcf74900b54355f109/.github/workflows/update-dependencies.yml#L16

---

Scratch that... https://github.com/actions/setup-python/issues/177 You're absolutely right.

---

Already forgot about the one you mentioned :-)